packages/proxy-target/contracts/HifiProxyTarget.sol

```diff
@@ -405,16 +393,22 @@ contract HifiProxyTarget is IHifiProxyTarget {
         hifiPool.underlying().safeTransfer(msg.sender, underlyingReturned);
 
         // Ensure that we are within the user's slippage tolerance.
-        uint256 underlyingOut = hifiPool.getQuoteForSellingUnderlying(hTokenReturned);
+        uint256 underlyingOut = hifiPool.getQuoteForSellingHToken(hTokenReturned);
```

Comment:
Wouldn't this revert if the pool was expired?

Reply:
Perhaps there needs to be two functions:
* `removeLiquidityAndSellHToken`
* `removeLiquidityAndRedeemHToken`

The FE can determine which one to call.  Otherwise if the pool is expired, then the `minUnderlyingOut` input param has no meaning in `removeLiquidityAndSellHToken`

Reply:
I agree, I'll add removeLiquidityAndRedeemHToken.

Reply:
Good catch with the pool's expiration, @defi-david.